src/main/java/com/mastery/java/task/dao/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,190 @@
+            preparedStatement.setString(1, employee.getFirstName());
+            preparedStatement.setString(2, employee.getLastName());
+            preparedStatement.setInt(3, employee.getDepartamentId());
+            preparedStatement.setString(4, employee.getJobTitle());
+            preparedStatement.setString(5, String.valueOf(employee.getGender()));
+            preparedStatement.setDate(6, (Date) employee.getDateOfBirth());
+
+            int out = preparedStatement.executeUpdate();
+            if (out != 0) {
+                System.out.println("Employee saved with id=" + employee.getEmployeeId());
+            } else System.out.println("Employee save failed with id=" + employee.getEmployeeId());
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                preparedStatement.close();
+                connection.close();
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+
+    }
+
+
+    public Employee getById(int id) {
+
+        String query = "SELECT employee_id, first_name, last_name, departament_id, job_title, gender, date_of_birth from employee where employee_id = ?";
+        Employee employee = null;
+        Connection con = null;
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+        try {
+            con = dataSource.getConnection();
+            preparedStatement = con.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                employee = new Employee();
+                employee.setEmployeeId(resultSet.getInt("employee_id"));
+                employee.setFirstName(resultSet.getString("first_name"));
+                employee.setLastName(resultSet.getString("last_name"));
+                employee.setDepartamentId(resultSet.getInt("departament_id"));
+                employee.setJobTitle(resultSet.getString("job_title"));
+                employee.setGender(Gender.valueOf(resultSet.getString("gender")));
+                employee.setDateOfBirth(resultSet.getDate("date_of_birth"));
+                System.out.println("Employee Found::" + employee);
+            } else {
+                System.out.println("No Employee found with id=" + id);
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                resultSet.close();
+                preparedStatement.close();
+                con.close();
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+        return employee;
+    }
+
+
+    public void update(Employee employee) {
+        String query = "UPDATE employee set first_name=?, last_name=?, departament_id=?, job_title=?, gender=?, date_of_birth=? where employee_id=?";
+        Connection con = null;
```

Comment:
Please use named constant String in the top of the class to store SQL query

Reply:
Done